src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1284,9 +1331,24 @@ static UCS_CLASS_INIT_FUNC(uct_dc_mlx5_iface_t, uct_md_h tl_md, uct_worker_h wor
         init_attr.flags  |= UCT_IB_TM_SUPPORTED;
     }
 
-    init_attr.cq_len[UCT_IB_DIR_TX] = config->super.super.tx.queue_len *
-                                      UCT_IB_MLX5_MAX_BB *
-                                      (config->ndci + UCT_DC_MLX5_KEEPALIVE_NUM_DCIS);
+    if (md->flags & UCT_IB_MLX5_MD_FLAG_DEVX_DCI) {
+        sq_length = uct_ib_mlx5_devx_sq_length(tx_queue_len);
+    } else {
+        if (md->dv_tx_wqe_ratio.dc == 0) {
+            /* TODO: calculate */
+            status = uct_dc_mlx5dv_calc_tx_wqe_ratio(md);
```

Comment:
can we make ratio to be an output param?

Reply:
done